### PR TITLE
Added time warning (6h)

### PR DESCRIPTION
**Pull request by UnzorZ**
I created a time warning. It sends a message to the player 10 times and plays a sound (anvil place with slight pitch change so its not that loud). Also, it keeps sending messages every minute.

### CODE REVIEW

**Comment by MrFums** [comment]
Hmm, doesnt seem to work for me?

**Comment by UnzorZ** [comment]
what is exactly not working? are you testing it on 2b2t?. It had been working for me for like 10 days or so

File: src/main/java/com/zenith/Proxy.java [hunk 1]
```diff
@@ -156,6 +156,35 @@ public void start() {
             // ensure we are continuously updating the tablist even on servers that don't frequently send updates
             SCHEDULED_EXECUTOR_SERVICE.scheduleAtFixedRate(this::tablistUpdate, 20L, 3L, TimeUnit.SECONDS);
             SCHEDULED_EXECUTOR_SERVICE.submit(this::updatePrioBanStatus);
+            // time warning
+            SCHEDULED_EXECUTOR_SERVICE.scheduleAtFixedRate(() -> {
```
Comment (rfresh2) [inline src/main/java/com/zenith/Proxy.java]: extract the runnable being submitted here to a method

File: src/main/java/com/zenith/Proxy.java [hunk 2]
```diff
@@ -156,6 +156,35 @@ public void start() {
...
+                try {
```
Comment (rfresh2) [inline src/main/java/com/zenith/Proxy.java]: add a check for if the account has prio or not in the runnable:

```
if (this.isPrio.orElse(false)) return; // skip the rest of the method and don't send any warnings
```

there's no 6hr kick for prio accounts

File: src/main/java/com/zenith/Proxy.java [hunk 3]
```diff
@@ -156,6 +156,35 @@ public void start() {
...
+                    if (isOnlineOn2b2tForAtLeastDuration(Duration.ofSeconds(60))) {
+                        long onlineSeconds = Instant.now().getEpochSecond() - connectTime.getEpochSecond();
+                        if (onlineSeconds > (21600 - (600))) { // 28800 - 600 (6h - 10 minutes)
+                            for (int i=0; i<10; i++) { //Sends the message 10 times, so you don't miss it. It also sends it every minute from then
```
Comment (rfresh2) [inline src/main/java/com/zenith/Proxy.java]: I think the amount of messages being sent is a bit excessive, I think just sending one every couple minutes would be more than enough.

Alternatively, we could set [the title, subtitle, or action bar texts](https://github.com/rfresh2/MCProtocolLib/tree/1.20/src/main/java/com/github/steveice10/mc/protocol/packet/ingame/clientbound/title). I'd personally prefer the action bar so its less obtrusive

File: src/main/java/com/zenith/Proxy.java [hunk 4]
```diff
@@ -156,6 +156,35 @@ public void start() {
...
+                                Proxy.getInstance().getActiveConnections().forEach(connection -> {
```
Comment (rfresh2) [inline src/main/java/com/zenith/Proxy.java]: just sending this to the active player is fine: 

```
final ServerConnection playerConnection = this.currentPlayer.get();
if (playerConnection != null) // we have a player logged in, send any warning packets to this connection
  playerConnection.send(...)
```

And it'd be better to just check if there is an active player or not near the top of this runnable to skip most of the logic if not.

`if (!this.isConnected() || !this.hasActivePlayer()) return; // no active player - no need to do any in-game warnings`

**Review by rfresh2 — CHANGES_REQUESTED** [review]

**Comment by MrFums** [comment]
> what is exactly not working? are you testing it on 2b2t?. It had been working for me for like 10 days or so

The "ding" noise is not coming through. Game sound maxed and what not, only thought could be the resource pack for some reason.

**Comment by UnzorZ** [comment]
If you think i should change something else, please let me know ^^

**Review by rfresh2 — APPROVED** [review]

**Comment by rfresh2** [comment]
thanks!

**Pull request merged**